tests/cli/test_orion.py

```diff
@@ -0,0 +1,51 @@
+from prefect.testing.cli import invoke_and_assert
+from prefect.testing.utilities import AsyncMock
+
+
+def test_orion_status_healthy():
+    invoke_and_assert(["orion", "status"], expected_output_contains="Healthy")
+
+
+def test_orion_status_unhealthy(monkeypatch):
+    async def mock_api_healthcheck(*args):
+        return Exception("All connection attempts failed")
+
+    monkeypatch.setattr(
+        "prefect.client.OrionClient.api_healthcheck", mock_api_healthcheck
+    )
+    invoke_and_assert(
+        ["orion", "status"],
+        expected_output_contains="All connection attempts failed",
+        expected_code=1,
+    )
+
+
+def test_orion_status_wait_arg(monkeypatch):
+    retry_response = "All connection attempts failed"
+    mock_waiting_healthcheck = AsyncMock()
+
+    monkeypatch.setattr(
+        "prefect.client.OrionClient.api_healthcheck", mock_waiting_healthcheck
+    )
+    mock_waiting_healthcheck.side_effect = [
+        retry_response,
+        retry_response,
+        retry_response,
+        None,
+    ]
+
+    invoke_and_assert(["orion", "status", "--wait"], expected_output_contains="Healthy")
+
+
+# Need to fix
+# def test_orion_status_timeout(monkeypatch):
+#     async def mock_api_healthcheck(*args):
+#         return Exception("All connection attempts failed")
+
+#     monkeypatch.setattr(
+#         "prefect.client.OrionClient.api_healthcheck", mock_api_healthcheck
+#     )
+#     invoke_and_assert(
+#         ["orion", "status", "--wait", "--timeout", 10],
+#         expected_output_contains="Timeout value exceeded"
+#     )
```

Comment:
@madkinsz I ran into some issues with this test since it doesn't seem to be respecting the anyio.move_on_after timeout, if you have any suggestions on this it would be greatly appreciated 😄

Reply:
I took a look and it should be working! 10 seconds is a while to run in a test. See https://github.com/PrefectHQ/prefect/pull/7970/commits/e5136122d25464cd1f3a90b08ef15be207899f43

Reply:
Awesome Thanks 😄